cmd/ignite-spawn/ignite-spawn.go

```diff
@@ -96,3 +100,18 @@ func patchStopped(vm *api.VM) error {
 	patch := []byte(`{"status":{"running":false,"network":null,"runtime":null,"startTime":null}}`)
 	return patchutil.NewPatcher(scheme.Serializer).ApplyOnFile(constants.IGNITE_SPAWN_VM_FILE_PATH, patch, vm.GroupVersionKind())
 }
+
+func parseEnvVars(vars []string) map[string]string {
```

Comment:
Since this is used just once, above, maybe `os.Environ()` should be called in this function itself for now. Keeping true to the meaning of the function name.

Reply:
IMO using map of config key=values without mutating the ignite-spawn Environment is better, but this is conflating a lot of config ideas.  (flags producing ENV vs. flags overriding ENV)

In general though, is this the proper strategy for config?

Since we already have the information in the VM object, ignite-spawn has access to it and it can read the value from it directly.

That will get of all of the flag-passing and parsing code within ignite & ignite-spawn.

nit:
The names of the variables/annotations in this code should not reference any idea of ENV.
We should call it what it is with a proper domain.
There should be an explicit key for each thing you want.
```yaml
metadata:
  annotations:
    ignite.weave.works/interface-count: '3'
    # ignite.weave.works/interface-mode/eth0: 'dhcp-bridge'
    # ignite.weave.works/interface-mode/eth2: 'tc-redirect'
    # ignite.weave.works/interface-mode/macvtap-uvnrt: 'macvtap'
```